src/schema/posts.ts

```diff
@@ -2390,6 +2363,69 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         ),
       }));
     },
+    deleteSourcePostModeration: async (
+      _,
+      { postId }: { postId: string },
+      ctx: AuthContext,
+    ): Promise<GQLEmptyResponse> => {
+      const moderation = await ctx.con
+        .getRepository(SourcePostModeration)
+        .findOneByOrFail({ id: postId });
+
+      await ensureSourcePermissions(ctx, moderation.sourceId);
+
+      const isModerator = await isPrivilegedMember(ctx, moderation.sourceId);
+      const isAuthor = moderation.createdById === ctx.userId;
+
+      if (!isModerator && !isAuthor) {
+        throw new ForbiddenError('Access denied!');
+      }
+
+      await ctx.con.getRepository(SourcePostModeration).delete({ id: postId });
+
+      return { _: true };
+    },
+    editSourcePostModeration: async (
+      _,
+      post: CreateSourcePostModerationArgs & { id: string },
+      ctx: AuthContext,
+      info,
+    ): Promise<SourcePostModeration> => {
+      const { id } = post;
+      const moderation = await ctx.con
+        .getRepository(SourcePostModeration)
+        .findOneByOrFail({ id });
+
+      const isAuthor = moderation.createdById === ctx.userId;
+      const isApproved =
+        moderation.status === SourcePostModerationStatus.Approved;
+
+      if (!isAuthor || isApproved) {
+        throw new ForbiddenError('Access denied!');
+      }
```

Comment:
Feels like `isApproved` should be ConflictError and separate?

Reply:
We can do validation error too since the request is not valid anymore.